Fail to delete the pipeline which has no pipeline records

**Environment**
- branch: v1-preview

**What happened**
Error when delete the pipeline which has no pipeline records:
`[2017-09-10 22:20:26] [ERROR] [pipeline.go:141] Fail to delete all pipeline records of the pipeline pipeline1 in the project project1 as not found`

**What you expected to happen**
- Correctly delete the pipeline which has no pipeline records.

--- a/pkg/server/manager/pipeline.go
+++ b/pkg/server/manager/pipeline.go
@@ -212,11 +212,6 @@ func (m *pipelineManager) ClearPipelinesOfProject(projectName string) error {
 func (m *pipelineManager) deletePipeline(pipeline *api.Pipeline) error {
 	ds := m.dataStore
 
-	service, err := ds.FindServiceByID(pipeline.ServiceID)
-	if err != nil {
-		return err
-	}
-
 	// Delete the versions related to this pipeline.
 	versions, err := ds.FindVersionsByServiceID(pipeline.ServiceID)
 	if err != nil {
@@ -229,31 +224,31 @@ func (m *pipelineManager) deletePipeline(pipeline *api.Pipeline) error {
 		}
 	}
 
-	// Delete the webhook.
-	remote, err := m.remoteManager.FindRemote(service.Repository.Webhook)
-	if err != nil {
-		logdog.Error(err.Error())
-	} else {
-		if err := remote.DeleteHook(service); err != nil {
-			logdog.Errorf("Fail to delete the webhook for pipeline %s as %s", pipeline.Name, err.Error())
+	if service, err := ds.FindServiceByID(pipeline.ServiceID); err == nil {
+		// Delete the webhook.
+		remote, err := m.remoteManager.FindRemote(service.Repository.Webhook)
+		if err != nil {
+			logdog.Error(err.Error())
+		} else {
+			if err := remote.DeleteHook(service); err != nil {
+				logdog.Errorf("Fail to delete the webhook for pipeline %s as %s", pipeline.Name, err.Error())
+			}
 		}
-	}
 
-	// Delete the service related to this pipeline
-	err = ds.DeleteServiceByID(pipeline.ServiceID)
-	if err != nil {
-		return fmt.Errorf("Fail to delete the service for pipeline %s as %s", pipeline.Name, err.Error())
+		// Delete the service related to this pipeline
+		err = ds.DeleteServiceByID(pipeline.ServiceID)
+		if err != nil {
+			return fmt.Errorf("Fail to delete the service for pipeline %s as %s", pipeline.Name, err.Error())
+		}
 	}
 
 	// Delete the pipeline records of this pipeline.
 	if err = m.pipelineRecordManager.ClearPipelineRecordsOfPipeline(pipeline.ID); err != nil {
-		logdog.Errorf("Fail to delete all pipeline records for pipeline %s as %s", pipeline.Name, err.Error())
-		return err
+		return fmt.Errorf("Fail to delete all pipeline records for pipeline %s as %s", pipeline.Name, err.Error())
 	}
 
 	if err = ds.DeletePipelineByID(pipeline.ID); err != nil {
-		logdog.Errorf("Fail to delete the pipeline %s as %s", pipeline.Name, err.Error())
-		return err
+		return fmt.Errorf("Fail to delete the pipeline %s as %s", pipeline.Name, err.Error())
 	}
 
 	return nil

--- a/store/pipelineRecord.go
+++ b/store/pipelineRecord.go
@@ -20,6 +20,7 @@ import (
 	"time"
 
 	"github.com/caicloud/cyclone/pkg/api"
+	"gopkg.in/mgo.v2"
 	"gopkg.in/mgo.v2/bson"
 )
 
@@ -87,5 +88,9 @@ func (d *DataStore) DeletePipelineRecordByID(pipelineRecordID string) error {
 
 // DeletePipelineRecordsByPipelineID deletes all the pipeline records of one pipeline by pipeline id.
 func (d *DataStore) DeletePipelineRecordsByPipelineID(pipelineID string) error {
-	return d.pipelineRecordCollection.Remove(bson.M{"pipelineId": pipelineID})
+	if err := d.pipelineRecordCollection.Remove(bson.M{"pipelineId": pipelineID}); err != mgo.ErrNotFound {
+		return err
+	}
+
+	return nil
 }

